lib/modules/manager/asdf/upgradeable-tooling.ts

```diff
@@ -0,0 +1,271 @@
+import { AdoptiumJavaDatasource } from '../../datasource/adoptium-java';
+import { DockerDatasource } from '../../datasource/docker';
+import { GithubReleasesDatasource } from '../../datasource/github-releases';
+import { GithubTagsDatasource } from '../../datasource/github-tags';
+import { NodeDatasource } from '../../datasource/node';
```

Comment:
please ensure they match the supported datasources export inside index file

Reply:
@viceice Done. I also added a test for it.

Reply:
please read the contribution guide

https://github.com/renovatebot/renovate/blob/main/.github/contributing.md#resolve-review-comments-instead-of-commenting